plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/AbstractIcebergTableOperations.java

```diff
@@ -143,7 +146,15 @@ public void commit(@Nullable TableMetadata base, TableMetadata metadata)
         }
 
         if (base == null) {
-            commitNewTable(metadata);
+            if (PROVIDER_PROPERTY_VALUE.equals(metadata.properties().get(PROVIDER_PROPERTY_KEY))) {
```

Comment:
The condition looks very tricky, doesn't it?

Reply:
That's exactly tricky. If we want to avoid this, I suppose it will require many duplicated codes though.

Reply:
pinged @electrum offline for his opinion